cmd/sgbouncewizard/main.go

```diff
@@ -55,12 +136,125 @@ func ListRules(w http.ResponseWriter, r *http.Request) {
 
 func generateHash(pwd []byte) string {
 	hash, err := bcrypt.GenerateFromPassword(pwd, bcrypt.MinCost)
+	if err != nil {
+		log.Println(err)
+		return "error"
+	}
+	return string(hash)
+}
+
+func createRuleDB(rule *BounceRule) []byte {
+	db, err := sql.Open("mysql", "root:root@tcp(127.0.0.1:3306)/drop_rules")
+	checkErr(err)
+
+	stmt, err := db.Prepare("INSERT INTO bounce_rule(id,response_code,enhanced_code,regex,priority,description,bounce_action) VALUES(?,?,?,?,?,?,?)")
+	checkErr(err)
+
+	_, err = stmt.Query(rule.Id, rule.Response_code, rule.Enhanced_code, rule.Regex, rule.Priority, rule.Description, rule.Bounce_action)
+	checkErr(err)
+
+	defer stmt.Close()
+	defer db.Close()
+
+	data, err := json.Marshal(rule)
+	checkErr(err)
+	return data
+}
+
+func getRuleDB(id string) (*BounceRule, error) {
+	db, err := sql.Open("mysql", "root:root@tcp(127.0.0.1:3306)/drop_rules")
+	var bounce_rule *BounceRule
+
+	checkErr(err)
+	rows, err := db.Query("SELECT * FROM bounce_rule WHERE id=" + id)
+
+	for rows.Next() {
+		var br BounceRule
+		err = rows.Scan(&br.Id, &br.Response_code, &br.Enhanced_code, &br.Regex, &br.Priority, &br.Description, &br.Bounce_action)
+		checkErr(err)
+		bounce_rule = &br
+		return bounce_rule, nil
+	}
+	checkErr(err)
+
+	defer rows.Close()
+	defer db.Close()
+	return nil, err
+}
 
+func deleteRuleDB(id int) (*BounceRule, error) {
+	db, err := sql.Open("mysql", "root:root@tcp(127.0.0.1:3306)/drop_rules")
+	checkErr(err)
+	query := fmt.Sprintf("%s%d", "DELETE FROM bounce_rule WHERE id=", id)
```

Comment:
While the use of sprintf here for this statement is _probably_ safe, generally try avoid `fmt.Sprintf` when creating queries to avoid sql injection. Instead consider:
```
_, err := db.Exec("DELETE FROM bounce_rule WHERE id = ?")
if err != nil{
    return nil, err
}
```